core/matching/orderbook.go

```diff
@@ -631,6 +631,32 @@ func (b *OrderBook) CancelAllOrders(party string) ([]*types.OrderCancellationCon
 	return confs, err
 }
 
+func (b *OrderBook) CheckBook() bool {
+	if len(b.buy.levels) > 0 {
+		allPegged := true
+		for _, o := range b.buy.levels[len(b.buy.levels)-1].orders {
+			if o.PeggedOrder == nil || o.PeggedOrder.Reference != types.PeggedReferenceBestBid {
+				allPegged = false
```

Comment:
Could do with a `break` here so we can save some time and stop looking.

Reply:
sure